bot/exts/fun/off_topic_names.py

```diff
@@ -51,24 +60,129 @@ async def update_names(self) -> None:
         """Background updater task that performs the daily channel name update."""
         await self.bot.wait_until_guild_available()
 
-        try:
-            channel_0_name, channel_1_name, channel_2_name = await self.bot.api_client.get(
-                "bot/off-topic-channel-names", params={"random_items": 3}
-            )
-        except ResponseCodeError as e:
-            log.error(f"Failed to get new off-topic channel names: code {e.response.status}")
-            raise
-
-        channel_0, channel_1, channel_2 = (self.bot.get_channel(channel_id) for channel_id in CHANNELS)
-
-        await channel_0.edit(name=OTN_FORMATTER.format(number=0, name=channel_0_name))
-        await channel_1.edit(name=OTN_FORMATTER.format(number=1, name=channel_1_name))
-        await channel_2.edit(name=OTN_FORMATTER.format(number=2, name=channel_2_name))
+        channels = [await get_or_fetch_channel(channel_id) for channel_id in CHANNELS]
+        failed_renames = defaultdict(list)
+        successfully_renamed = set()
+        for attempt in range(MAX_RENAME_ATTEMPTS):
+            # Get the necessary amount of new channel names
```

Comment:
I don't think that all these comments are necessary, comments should exist to document things that don't jump straight to the eye.

Reply:
I don't see any real harm in documenting how the code works given the complexity.

Reply:
That's what I'm saying, that the code can be understood by simply reading it, as the logic is kinda straightforward.

I personally don't think it's complex, we'll see what others have to say about it.

Reply:
Yeah, in hindsight I agree -- these comments were removed as part of the rewrite.